### ISSUE TITLE
Application hello error: capture stdout: EOF

### ISSUE THREAD

**Issue by olezhek28** [ServiceWeaver/weaver#302]
Hello! I tried to run hello example, but had error.

`weaver multi deploy weaver.toml`
`Application hello error: capture stdout: EOF`

I used Apple Macbook M1.

**weaver.toml**
```
[serviceweaver]
binary = "./hello"
```

**main.go**
```
package main

import (
	"context"
	"fmt"
	"net/http"

	"github.com/ServiceWeaver/weaver"
)

func main() {
	weaver.Run(context.Background(), serve)
}

//go:generate ../../cmd/weaver/weaver generate

type server struct {
	weaver.Implements[weaver.Main]
}

func serve(ctx context.Context, s *server) error {
	// Get a client to the Reverser component.
	reverser, err := weaver.Get[Reverser](s)
	if err != nil {
		return err
	}

	// Get a network listener on address "localhost:12345".
	opts := weaver.ListenerOptions{LocalAddress: "localhost:12345"}
	lis, err := s.Listener("hello", opts)
	if err != nil {
		return err
	}
	fmt.Printf("hello listener available on %v\n", lis)

	// Serve the /hello endpoint.
	http.Handle("/hello", weaver.InstrumentHandlerFunc("hello",
		func(w http.ResponseWriter, r *http.Request) {
			reversed, err := reverser.Reverse(r.Context(), r.URL.Query().Get("name"))
			if err != nil {
				http.Error(w, err.Error(), http.StatusInternalServerError)
				return
			}
			fmt.Fprintf(w, "Hello, %s!\n", reversed)
		}))
	return http.Serve(lis, nil)
}

```

**reverser.go**
```
package main

import (
	"context"

	"github.com/ServiceWeaver/weaver"
)

// Reverser component.
type Reverser interface {
	Reverse(context.Context, string) (string, error)
}

// Implementation of the Reverser component.
type reverser struct {
	weaver.Implements[Reverser]
}

func (r *reverser) Reverse(_ context.Context, s string) (string, error) {
	runes := []rune(s)
	n := len(runes)
	for i := 0; i < n/2; i++ {
		runes[i], runes[n-i-1] = runes[n-i-1], runes[i]
	}
	return string(runes), nil
}
```

**Comment by mwhittaker** [comment]
Hi @olezhek28, can you do the following. 

1. Share the output of `weaver multi version`. 
2. Share the version of the weaver module in your `go.mod` file? 
3. Run `weaver multi purge`.
4. Re-run the app. 
5. After it crashes, share the output of `weaver multi logs --system`.

**Comment by olezhek28** [comment]
1. weaver multi: commit=? deployer=v0.6.0 target=darwin/arm64 version
2. 
```
module service_weaver

go 1.20

require github.com/ServiceWeaver/weaver v0.6.1

require (
	github.com/BurntSushi/toml v1.2.0 // indirect
	github.com/DataDog/hyperloglog v0.0.0-20220214164406-974598347557 // indirect
	github.com/antlr/antlr4/runtime/Go/antlr v0.0.0-20220418222510-f25a4f6275ed // indirect
	github.com/dustin/go-humanize v1.0.0 // indirect
	github.com/fsnotify/fsnotify v1.5.4 // indirect
	github.com/go-logr/logr v1.2.3 // indirect
	github.com/go-logr/stdr v1.2.2 // indirect
	github.com/google/cel-go v0.12.5 // indirect
	github.com/google/pprof v0.0.0-20221118152302-e6195bd50e26 // indirect
	github.com/google/uuid v1.3.0 // indirect
	github.com/hashicorp/golang-lru/v2 v2.0.1 // indirect
	github.com/kballard/go-shellquote v0.0.0-20180428030007-95032a82bc51 // indirect
	github.com/lightstep/varopt v1.3.0 // indirect
	github.com/mattn/go-isatty v0.0.16 // indirect
	github.com/pkg/browser v0.0.0-20210911075715-681adbf594b8 // indirect
	github.com/remyoudompheng/bigfft v0.0.0-20230129092748-24d4a6f8daec // indirect
	github.com/stoewer/go-strcase v1.2.0 // indirect
	go.opentelemetry.io/otel v1.13.0 // indirect
	go.opentelemetry.io/otel/exporters/stdout/stdouttrace v1.7.0 // indirect
	go.opentelemetry.io/otel/sdk v1.11.1 // indirect
	go.opentelemetry.io/otel/trace v1.13.0 // indirect
	golang.org/x/exp v0.0.0-20230321023759-10a507213a29 // indirect
	golang.org/x/mod v0.6.0 // indirect
	golang.org/x/sync v0.0.0-20220722155255-886fb9371eb4 // indirect
	golang.org/x/sys v0.1.0 // indirect
	golang.org/x/term v0.0.0-20210927222741-03fcf44c2211 // indirect
	golang.org/x/text v0.7.0 // indirect
	golang.org/x/tools v0.2.0 // indirect
	google.golang.org/genproto v0.0.0-20221109142239-94d6d90a7d66 // indirect
	google.golang.org/protobuf v1.28.1 // indirect
	lukechampine.com/uint128 v1.2.0 // indirect
	modernc.org/cc/v3 v3.40.0 // indirect
	modernc.org/ccgo/v3 v3.16.13 // indirect
	modernc.org/libc v1.22.3 // indirect
	modernc.org/mathutil v1.5.0 // indirect
	modernc.org/memory v1.5.0 // indirect
	modernc.org/opt v0.1.3 // indirect
	modernc.org/sqlite v1.21.0 // indirect
	modernc.org/strutil v1.1.3 // indirect
	modernc.org/token v1.0.1 // indirect
)

```

3. 
```
weaver multi purge
WARNING: You are about to kill all processes which match the following regex:                                                                                                                                                                 

    weaver multi (dashboard|deploy|logs|profile)

This currently includes the following processes:

        (no matching processes found)
    
You will also delete the following paths used to store logs and data for
"weaver multi" Service Weaver applications. This data will be deleted
immediately and irrevocably. Are you sure you want to proceed?"

    - /var/folders/rq/dxk00vbd56q8qn7dv8f3lzn40000gp/T/serviceweaver/logs/weaver-multi
    - /Users/okozyrev/.local/share/serviceweaver/multi_registry
    - /Users/okozyrev/.local/share/serviceweaver/perfetto_multi.db

Enter (y)es to continue: y

exit status 2: pkill: illegal option -- e
usage: pkill [-signal] [-ILfilnovx] [-F pidfile] [-G gid]
             [-P ppid] [-U uid] [-g pgrp] [-t tty] [-u euid]
             pattern ...

```

4. success

5.
```
D0502 10:37:09.904803 weavelet 0368374f weavelet.go:297] i logs --system                                                                                                                                                        1 ✘  took 44s   at 10:37:54  
┌ weavelet 0368374f-4d6d-4e90-967a-1bd2baa8a178 started ┐
│   hostname   : mbp-okozyrev-OZON-FVFFK3TZQ05N         │
│   deployment : 301aab27-7f0b-4d60-91e7-753ff1151a49   │
│   address    : tcp://127.0.0.1:58589                  │
│   pid        : 21452                                  │
└───────────────────────────────────────────────────────┘ serviceweaver/system=""
D0502 10:37:09.905121 weavelet 0368374f weavelet.go:307] Registering component... component="github.com/ServiceWeaver/weaver/Main" serviceweaver/system=""
D0502 10:37:09.905093 weavelet 0368374f weavelet.go:429] UpdateComponents components="[github.com/ServiceWeaver/weaver/Main]" serviceweaver/system=""
D0502 10:37:09.905267 weavelet 0368374f weavelet.go:534] Constructing component component="github.com/ServiceWeaver/weaver/Main" serviceweaver/system=""
D0502 10:37:09.905315 weavelet 0368374f weavelet.go:539] Constructing component succeeded component="github.com/ServiceWeaver/weaver/Main" serviceweaver/system=""
D0502 10:37:09.923406 weavelet 00dbbed7 weavelet.go:297] 
┌ weavelet 00dbbed7-d330-40d0-a4be-13729bd365ee started ┐
│   hostname   : mbp-okozyrev-OZON-FVFFK3TZQ05N         │
│   deployment : 301aab27-7f0b-4d60-91e7-753ff1151a49   │
│   address    : tcp://127.0.0.1:58590                  │
│   pid        : 21453                                  │
└───────────────────────────────────────────────────────┘ serviceweaver/system=""
D0502 10:37:09.923448 weavelet 00dbbed7 weavelet.go:307] Registering component... component="github.com/ServiceWeaver/weaver/Main" serviceweaver/system=""
D0502 10:37:09.923567 weavelet 00dbbed7 weavelet.go:429] UpdateComponents components="[github.com/ServiceWeaver/weaver/Main]" serviceweaver/system=""
D0502 10:37:09.923609 weavelet 00dbbed7 weavelet.go:534] Constructing component component="github.com/ServiceWeaver/weaver/Main" serviceweaver/system=""
D0502 10:37:09.923631 weavelet 00dbbed7 weavelet.go:539] Constructing component succeeded component="github.com/ServiceWeaver/weaver/Main" serviceweaver/system=""
D0502 10:37:09.923868 weavelet 0368374f weavelet.go:456] Updating routing info... component="github.com/ServiceWeaver/weaver/Main" local="true" replicas="[]" serviceweaver/system=""
D0502 10:37:09.923883 weavelet 0368374f weavelet.go:461] Updating routing info succeeded component="github.com/ServiceWeaver/weaver/Main" local="true" replicas="[]" serviceweaver/system=""
D0502 10:37:09.923958 weavelet 0368374f weavelet.go:315] Registering component succeeded component="github.com/ServiceWeaver/weaver/Main" serviceweaver/system=""
D0502 10:37:09.923966 weavelet 0368374f weavelet.go:307] Registering component... component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.923984 weavelet 00dbbed7 weavelet.go:456] Updating routing info... component="github.com/ServiceWeaver/weaver/Main" local="true" replicas="[]" serviceweaver/system=""
D0502 10:37:09.923993 weavelet 00dbbed7 weavelet.go:461] Updating routing info succeeded component="github.com/ServiceWeaver/weaver/Main" local="true" replicas="[]" serviceweaver/system=""
D0502 10:37:09.924037 weavelet 00dbbed7 weavelet.go:315] Registering component succeeded component="github.com/ServiceWeaver/weaver/Main" serviceweaver/system=""
D0502 10:37:09.924045 weavelet 00dbbed7 weavelet.go:307] Registering component... component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.924105 weavelet 0368374f weavelet.go:456] Updating routing info... component="service_weaver/hello/Reverser" local="false" replicas="[]" serviceweaver/system=""
D0502 10:37:09.924117 weavelet 0368374f weavelet.go:461] Updating routing info succeeded component="service_weaver/hello/Reverser" local="false" replicas="[]" serviceweaver/system=""
D0502 10:37:09.924153 weavelet 0368374f weavelet.go:456] Updating routing info... component="service_weaver/hello/Reverser" local="false" replicas="[]" serviceweaver/system=""
D0502 10:37:09.924162 weavelet 0368374f weavelet.go:461] Updating routing info succeeded component="service_weaver/hello/Reverser" local="false" replicas="[]" serviceweaver/system=""
D0502 10:37:09.939392 weavelet cf532938 weavelet.go:297] 
┌ weavelet cf532938-a384-4126-b14f-7e891f8bb880 started ┐
│   hostname   : mbp-okozyrev-OZON-FVFFK3TZQ05N         │
│   deployment : 301aab27-7f0b-4d60-91e7-753ff1151a49   │
│   address    : tcp://127.0.0.1:58592                  │
│   pid        : 21454                                  │
└───────────────────────────────────────────────────────┘ serviceweaver/system=""
D0502 10:37:09.939406 weavelet 0368374f weavelet.go:456] Updating routing info... component="service_weaver/hello/Reverser" local="false" replicas="[tcp://127.0.0.1:58592]" serviceweaver/system=""
D0502 10:37:09.939422 weavelet 0368374f weavelet.go:461] Updating routing info succeeded component="service_weaver/hello/Reverser" local="false" replicas="[tcp://127.0.0.1:58592]" serviceweaver/system=""
D0502 10:37:09.939537 weavelet cf532938 weavelet.go:429] UpdateComponents components="[service_weaver/hello/Reverser]" serviceweaver/system=""
D0502 10:37:09.939574 weavelet cf532938 weavelet.go:534] Constructing component component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.939605 weavelet cf532938 weavelet.go:539] Constructing component succeeded component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.952354 weavelet 7e810dd6 weavelet.go:297] 
┌ weavelet 7e810dd6-79f8-46b0-8925-246ba96df10f started ┐
│   hostname   : mbp-okozyrev-OZON-FVFFK3TZQ05N         │
│   deployment : 301aab27-7f0b-4d60-91e7-753ff1151a49   │
│   address    : tcp://127.0.0.1:58593                  │
│   pid        : 21455                                  │
└───────────────────────────────────────────────────────┘ serviceweaver/system=""
D0502 10:37:09.952461 weavelet 0368374f weavelet.go:456] Updating routing info... component="service_weaver/hello/Reverser" local="false" replicas="[tcp://127.0.0.1:58592 tcp://127.0.0.1:58593]" serviceweaver/system=""
D0502 10:37:09.952480 weavelet 0368374f weavelet.go:461] Updating routing info succeeded component="service_weaver/hello/Reverser" local="false" replicas="[tcp://127.0.0.1:58592 tcp://127.0.0.1:58593]" serviceweaver/system=""
D0502 10:37:09.952597 weavelet 7e810dd6 weavelet.go:429] UpdateComponents components="[service_weaver/hello/Reverser]" serviceweaver/system=""
D0502 10:37:09.952644 weavelet 7e810dd6 weavelet.go:534] Constructing component component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.952675 weavelet 7e810dd6 weavelet.go:539] Constructing component succeeded component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.952701 weavelet 0368374f weavelet.go:315] Registering component succeeded component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.952719 weavelet 0368374f weavelet.go:608] Creating a connection to a remote component... component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.952810 weavelet 00dbbed7 weavelet.go:456] Updating routing info... component="service_weaver/hello/Reverser" local="false" replicas="[tcp://127.0.0.1:58592 tcp://127.0.0.1:58593]" serviceweaver/system=""
D0502 10:37:09.952839 weavelet 00dbbed7 weavelet.go:461] Updating routing info succeeded component="service_weaver/hello/Reverser" local="false" replicas="[tcp://127.0.0.1:58592 tcp://127.0.0.1:58593]" serviceweaver/system=""
D0502 10:37:09.952893 weavelet 00dbbed7 weavelet.go:315] Registering component succeeded component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.952906 weavelet 00dbbed7 weavelet.go:608] Creating a connection to a remote component... component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.959685 weavelet 0368374f weavelet.go:623] Creating connection to remote component succeeded component="service_weaver/hello/Reverser" serviceweaver/system=""
D0502 10:37:09.959690 weavelet 00dbbed7 weavelet.go:623] Creating connection to remote component succeeded component="service_weaver/hello/Reverser" serviceweaver/system=""
I0502 10:37:09.961280 deployer 89900713 deployer.go:465] Proxy listening address="127.0.0.1:12345" serviceweaver/system=""
S0502 10:37:09.961581 stdout   0368374f                ] hello listener available on 127.0.0.1:12345
S0502 10:37:09.961673 stdout   00dbbed7                ] hello listener available on 127.0.0.1:12345
```

**Comment by mwhittaker** [comment]
Ah, sweet. It looks like things are working then? I'll close the issue, but please re-open if I missed something. It also looks like the `-e` flag we're passing to `pkill` doesn't work on mac. I'll look into that.
